Mathlib/LinearAlgebra/FiniteDimensional/Basic.lean

```diff
@@ -393,6 +393,22 @@ theorem ofInjectiveEndo_left_inv (f : V →ₗ[K] V) (h_inj : Injective f) :
     ((ofInjectiveEndo f h_inj).symm : V →ₗ[K] V) * f = 1 :=
   LinearMap.ext <| (ofInjectiveEndo f h_inj).symm_apply_apply
 
+variable {V' : Type*} [AddCommGroup V'] [Module K V'] [FiniteDimensional K V']
+omit [FiniteDimensional K V]
+
+/-- An injective linear map between finite-dimensional space of equal rank
+is a linear equivalence. -/
+noncomputable def ofInjectiveOfFinrankEq (f : V →ₗ[K] V') (hinj : Function.Injective f)
```

Comment:
Can you explain in the docstring how this differs from `LinearEquiv.ofFinrankEq`?

Reply:
Good catch. The answer is yes and no. Yes - strictly speaking, finite-dimensionality of V is not assumed; on the other hand, this is implied by the finrank equality, V' being finite-dimensional and f being injective. (At least if F is a field; in general, I'm not sure if there is a catch, I'd need to review by ring theory to make a confident statement.) I chose to omit this simply as there's less to prove --- but mentioning that is not mathematically interesting, hence made a small lie in the doc-string.